go/tools/filter_tags/filter_tags.go

```diff
@@ -55,16 +60,76 @@ func filterFilenames(bctx build.Context, inputs []string) ([]string, error) {
 	return outputs, nil
 }
 
+// Returns an array of strings containing only the filenames that should build
+// according to the Context given.
+func runCommand(bctx build.Context, inputs []string) error {
+	if len(inputs) <= 0 {
+		return fmt.Errorf("filter_tags in run mode but with no command to run")
+	}
+	executable := inputs[0]
+	args := []string{}
+	unfiltered := 0
+	filtered := 0
+	for _, in := range inputs[1:] {
+		if strings.HasPrefix(in, absMarker) {
+			in, _ = filepath.Abs(in[len(absMarker):])
+		}
+		if !strings.HasPrefix(in, filterMarker) {
+			// not a filter candidate
+			args = append(args, in)
+			continue
+		}
+		filename := in[len(filterMarker):]
+		fullPath, err := filepath.Abs(filename)
+		if _, err := os.Stat(filename); err != nil {
+			// not a valid file, don't filter
+			args = append(args, filename)
+			continue
+		}
+		dir, base := filepath.Split(fullPath)
+		matches, err := bctx.MatchFile(dir, base)
```

Comment:
Not sure what you meant, but I have changed the flow so much I am not sure the comment would still apply anyway